test/e2e/simple/e2e_test.go

```diff
@@ -0,0 +1,248 @@
+// +build e2e
+
+package simple
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"testing"
+
+	"github.com/openshift/ci-tools/test/e2e/framework"
+)
+
+func TestSimpleExitCodes(t *testing.T) {
+	var testCases = []struct {
+		name    string
+		args    []string
+		success bool
+		output  []string
+	}{
+		{
+			name:    "success on one successful target",
+			args:    []string{"--target=success"},
+			success: true,
+			output:  []string{"Container test in pod success completed successfully"},
+		},
+		{
+			name:    "failure on one successful and one failed target",
+			args:    []string{"--target=success", "--target=failure"},
+			success: false,
+			output:  []string{"Container test in pod success completed successfully", "Container test in pod failure failed, exit code 1, reason Error"},
+		},
+		{
+			name:    "failure on one failed target",
+			args:    []string{"--target=failure"},
+			success: false,
+			output:  []string{"Container test in pod failure failed, exit code 1, reason Error"},
+		},
+	}
+
+	for _, tc := range testCases {
+		testCase := tc
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			testDone := make(chan struct{})
+			cleanupDone := make(chan struct{})
+			defer func() {
+				close(testDone)
+				<-cleanupDone
+			}()
+			cmd, _ := framework.CiOperatorCommand(t)
+			cmd.Args = append(cmd.Args, append(testCase.args, "--config=config.yaml")...)
+			cmd.Env = append(cmd.Env, `JOB_SPEC={"type":"postsubmit","job":"branch-ci-openshift-ci-tools-master-ci-operator-e2e","buildid":"0","prowjobid":"uuid","refs":{"org":"openshift","repo":"ci-tools","base_ref":"master","base_sha":"6d231cc37652e85e0f0e25c21088b73d644d89ad","pulls":[]}}`)
+			output, err := framework.Execute(t, cmd, testDone, cleanupDone)
+			if testCase.success != (err == nil) {
+				t.Fatalf("%s: didn't expect an error from ci-operator: %v; output:\n%v", testCase.name, err, string(output))
```

Comment:
I think we should test for the precise failure we are expecting and not any failure to prevent these tests from rotting

Reply:
We do, literally two lines down we check for precise messages for that reason.